### PR TITLE
add sorting on back

### PR DESCRIPTION
**Pull request by Olesya987**

### CODE REVIEW

File: src/modules/controllers/appointment.controllers.js [hunk 1]
```diff
@@ -31,6 +31,33 @@ module.exports.getPag = (req, res) => {
     });
 };
 
+module.exports.sortAppointment = (req, res) => {
+  const { params, body } = req;
+  const tokenUser = parseJwt(req.headers.authorization);
+  const rowsOnPage = +params.rowsOnPage || 5;
+  const currentPage = +params.currentPage || 1;
+  const value = body.value;
+  const direction = +body.direction;
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: it is number?

Comment (Olesya987) [reply src/modules/controllers/appointment.controllers.js]: yes

File: src/modules/controllers/appointment.controllers.js [hunk 2]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
-    .then((appointments) => {
-      Appointment.count({ userId: tokenUser._id }).then((result) => {
-        res.send({
-          allRows: result,
-          appointments,
+  if (
+    body.hasOwnProperty("value") &&
+    body.hasOwnProperty("direction") &&
+    body.hasOwnProperty("before") &&
+    body.hasOwnProperty("after") &&
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js:20]: body.hasOwnProperty("after") &&
    body.value &&
    body.direction &&
    body.before &&
    body.after

File: src/modules/controllers/appointment.controllers.js [hunk 3]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
+    body.value.length !== 0 &&
+    body.direction.length !== 0 &&
+    body.before.length !== 0 &&
+    body.after.length !== 0
+  ) {
+    const value = body.value;
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: destructurization

File: src/modules/controllers/appointment.controllers.js [hunk 4]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
+            appointments,
+          });
         });
       });
-    });
-};
+  } else if (
+    body.hasOwnProperty("value") &&
+    body.hasOwnProperty("direction") &&
+    body.value.length !== 0 &&
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: higher

File: src/modules/controllers/appointment.controllers.js [hunk 5]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
+    body.direction.length !== 0
+  ) {
+    const value = body.value;
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: destructurization

File: src/modules/controllers/appointment.controllers.js [hunk 6]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
+          res.send({
+            allRows: result,
+            appointments,
+          });
+        });
+      });
+  } else if (
+    body.hasOwnProperty("before") &&
+    body.hasOwnProperty("after") &&
+    body.before.length !== 0 &&
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: higher

File: src/modules/controllers/appointment.controllers.js [hunk 7]
```diff
@@ -8,54 +8,114 @@ const parseJwt = (token) => {
...
+    body.after.length !== 0
+  ) {
+    const before = body.before;
```
Comment (usatovaDasha) [inline src/modules/controllers/appointment.controllers.js]: higher

**Pull request merged**